v2/internal/crdmanagement/crds.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package crdmanagement
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/Azure/azure-service-operator/v2/internal/util/kubeclient"
+)
+
+// ServiceOperatorVersionLabel is the label the CRDs have on them containing the ASO version. This value must match the value
+// injected by config/crd/labels.yaml
+const ServiceOperatorVersionLabel = "serviceoperator.azure.com/version"
+const CRDLocation = "crds"
+
+const certMgrInjectCAFromAnnotation = "cert-manager.io/inject-ca-from"
+
+// TODO: Fix logging levels in this whole file
+
+type Manager struct {
+	logger     logr.Logger
+	kubeClient kubeclient.Client
+}
+
+func NewManager(logger logr.Logger, kubeClient kubeclient.Client) *Manager {
+	return &Manager{
+		logger:     logger,
+		kubeClient: kubeClient,
+	}
+}
+
+func (m *Manager) ListOperatorCRDs(ctx context.Context) ([]apiextensions.CustomResourceDefinition, error) {
+	list := apiextensions.CustomResourceDefinitionList{}
+
+	selector := labels.NewSelector()
+	requirement, err := labels.NewRequirement(ServiceOperatorVersionLabel, selection.Exists, nil)
+	if err != nil {
+		return nil, err
+	}
+	selector = selector.Add(*requirement)
+
+	match := client.MatchingLabelsSelector{
+		Selector: selector,
+	}
+	err = m.kubeClient.List(ctx, &list, match)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list CRDs")
+	}
+
+	for _, crd := range list.Items {
+		m.logger.V(0).Info("Found a CRD", "CRD", crd.Name)
+	}
+
+	return list.Items, nil
+}
+
+func (m *Manager) LoadOperatorCRDs(path string) ([]apiextensions.CustomResourceDefinition, error) {
+	// Expectation is that every file in this folder is a CRD
+	entries, err := os.ReadDir(path)
```

Comment:
Should you ~dig~ recurse into nested folders so that we've got an option later on to organize the CRDs in some way, or is that a case of YAGNI?